src/SFA.DAS.EAS.Portal/Application/Commands/Reservation/AddReservationCommand.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Database.Models;
+using SFA.DAS.Reservations.Messages;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Reservation
+{
+    public class AddReservationCommand
+    {
+        private readonly IAccountDocumentService _accountsService;
+        private readonly ILogger<AddReservationCommand> _logger;
+
+        public AddReservationCommand(IAccountDocumentService accountsService, ILogger<AddReservationCommand> logger)
+        {
+            _accountsService = accountsService;
+            _logger = logger;
+        }
+
+        public async Task Execute(ReservationCreatedEvent reservedFunding, string messageId, CancellationToken cancellationToken = default)
+        {
+            // TODO: move logging to a decorator
+            _logger.LogInformation("Executing AddReservationCommand");
+
+            var accountDocument = await _accountsService.Get(reservedFunding.AccountId);
```

Comment:
The `cancellationToken` should be passed in here.

Reply:
Good spot.  Done.